app/products/calls/actions/calls.ts

```diff
@@ -94,24 +97,7 @@ export const loadCalls = async (serverUrl: string, userId: string) => {
 
     for (const channel of resp) {
         if (channel.call) {
-            const call = channel.call;
-            callsResults[channel.channel_id] = {
-                participants: channel.call.users.reduce((accum, cur, curIdx) => {
-                    // Add the id to the set of UserModels we want to ensure are loaded.
-                    ids.add(cur);
-
-                    // Create the CallParticipant
-                    const muted = call.states && call.states[curIdx] ? !call.states[curIdx].unmuted : true;
-                    const raisedHand = call.states && call.states[curIdx] ? call.states[curIdx].raised_hand : 0;
-                    accum[cur] = {id: cur, muted, raisedHand};
-                    return accum;
-                }, {} as Dictionary<CallParticipant>),
-                channelId: channel.channel_id,
-                startTime: call.start_at,
-                screenOn: call.screen_sharing_id,
-                threadId: call.thread_id,
-                ownerId: call.owner_id,
-            };
+            callsResults[channel.channel_id] = createCallAndAddToIds(channel.channel_id, channel.call, ids);
```

Comment:
Not totally against it but do we really need the complexity of the side effect? Could we use the returned information instead to retrieve the ids?

Reply:
@streamer45 thanks for bringing that up. We were originally doing as you suggest, but moved to adding to the `ids` `Set` within the loop, to address previous PR comments. At least this way we avoid looping through the participants object twice (times the number of active calls). 
I'm always wary of side effects like this, that's why I added it to the function name.

Reply:
Thanks for the context. Wouldn't it be sufficient to extend the call object to also have a participantsIds or something like that?

Reply:
@streamer45 We could... But I don't think it belongs there--we only need the list of ids locally, to assemble a list of user profiles to fetch. We could return a modified `Set` (but actually the original would still be mutated, and I don't want to waste resources making copies...). Hmm... Maybe make an array and return it as part of a tuple, then add the array to the set manually?

Reply:
I see, let's leave it as is for now. Honestly it feels a bit like premature optimization considering it doesn't look like a particularly hot path. With that said, maybe we could improve things on the server side by returning exactly what's needed by the client and that in turn could simplify the logic a bit.

Reply:
@streamer45 Cool, I like the idea of normalizing what we send to the client -- you can see we do some transformation to turn it into what we need. But maybe we should do an overhaul of what we store on the server and how, first. That seems like a win too.